### ISSUE TITLE
Feature request: dataset over multiple NetCDF files.

### ISSUE THREAD

**Issue by briochemc** [JuliaDataCubes/YAXArrays.jl#461]
How does one open a dataset spread over multiple files (like with xarray's `open_mfdataset` function)?

**Comment by Balinus** [comment]
There is an undocumented function:

https://github.com/JuliaDataCubes/YAXArrays.jl/blob/04aaaad774005ee8502a08cd195ab1b0d16b87e3/src/DatasetAPI/Datasets.jl#L335
and
https://github.com/JuliaDataCubes/YAXArrays.jl/blob/04aaaad774005ee8502a08cd195ab1b0d16b87e3/src/DatasetAPI/Datasets.jl#L945

I do not remember exactly how to use it, something along the lines of:

`YAXArrays.Datasets.open_mfdataset("/path/files")` or perhaps `YAXArrays.Datasets.open_mfdataset(list_of_files)` ? or both

**Comment by Balinus** [comment]
I had to reuse some code where mfdataset was used. In this case, it worked as expected. 

![image](https://github.com/user-attachments/assets/ef5dd6f0-964d-4eb9-bc72-234c016bebdd)

**Comment by briochemc** [comment]
Yes I've used it successfully too 😃 

I think it ought to be documented though, right?

**Comment by Balinus** [comment]
Is there a `preprocessing` option that can be added to `open_mfdataset`? I'd like to add a time dimension based on the filename for a suite of > 300 files that has only 2D longitude-latitude data.

for example, in Python I can do:

```python
import xarray as xr
import numpy as np


def preprocessing(ds): 
    time_str = os.path.basename(ds.encoding['source']).split('_')[0:2]    # get year and month from filename
    time_str.append('01') # add a hardcoded day to list of time
    time_str =  "-".join(time_str) # create a single date string
    ds['time'] = np.datetime64(time_str, 'ns') # assign to dataset ds
    ds = ds.set_coords('time') # specify time as a coordinate
    return ds

ds_crps = xr.open_mfdataset(list_of_files, concat_dim='time', combine='nested', preprocess = preprocessing)
```

**Comment by lazarusA** [comment]
this should be fixed by #470, #481
